jgrapht-core/src/main/java/org/jgrapht/traverse/LexBreadthFirstIterator.java

```diff
@@ -379,4 +419,34 @@ boolean isEmpty() {
             }
         }
     }
+    
+    class PriorityComparator implements Comparator<V>
+    {
+        private final V[] priority;
```

Comment:
As @pdelvo mentioned it a commit before, a HashMap should be better here to decrease the compare complexity